BLOOD FOR THE BLOOD GOD

paprka opened the pull request:
Adds blood.
Adds bleending.
Adds blood packs.
Adds IV drips.
Brute damage dictates how much blood you lose every drip. 
Low blood count will cause you to pass out sometimes and also die if you have no blood. 
Blood is produced faster the more nutriment and iron you have in your diet.
Readds bruise packs and ointment to medkits.
Renames some medkits.
Reorganizes medical storage. (4 first aid kits, 3 of each specialized type)
Adds reinforced wals on the maint side of medical storage.
Nerfs bruise packs and ointment to not apply through hardsuits.
Nerfs bruise packs and ointment to require a delay for applying.
Adds gauze to stop bleeding.

Thanks to bay12 for like, almost all of this code.
The baymed port has begun. Run for the hills.

Commit:
blood for the blood god

paprka commented:
>failed
>compiled on local

eric

Iamgoofball commented:
THE BLEEDENING BEGINS

@paprka can you make styptic and synthflesh stop bleeding while you're in the files

paprka commented:
yeah, i'll add gauze for medkits as well that basically do that

paprka commented:
pretty sure i used some /red inhere

JJRcop commented:
Don't worry about the travis thing I have it under control

Kromgar commented:
So how will this interact with cult? Because this could royally fuck cult if they have to run to medbay after drawing 5 runes from bleeding

paprka commented:
I'm going to add easier ways for people to patch brute damage, especially as medical doctors, the whole reliance on shitty chems that are broken is pretty bad

JJRcop commented:
a jumpsuit would be good too, since everyone has one

paprka commented:
I'll probably add a way to use wirecutters on a bed/jumpsuit to get ghetto gauze

Steelpoint commented:
The Bay/Goon Med change has begun.

I'll be interested in seeing how this works.

paprka commented:
Going to do a big update on this tomorrow with more ways to heal brute and ways to temporarily block bleeding

Steelpoint commented:
A good explanation of how this system works in its entirety would be great.

Lumipharon commented:
Just add bandages, and let people rip bedsheets/jumpsuits up (or say, a sleeve or w/e) for makeshift bandages.

Also I unno how that code works, but if brute damage effects how much you bleed, doesn't that uh, not make a lot of sense?
Like if someone stabs me in the eye with a screwdriver and I start bleeding, I probably wouldn't bleed worse if someone punched my legs to shit.
Or maybe I'm just misunderstanding. Either way, 10/10.

RemieRichards commented on code/modules/mob/living/carbon/human/blood.dm:
```diff
@@ -0,0 +1,310 @@
+					pale = 1
+					update_body()
+					var/word = pick("dizzy","woosey","faint")
+					src << "\red You feel [word]"
```
Muh span classes pls.

RemieRichards commented on code/modules/reagents/Chemistry-Holder.dm:
```diff
@@ -107,9 +107,15 @@ datum/reagents/proc/trans_to(var/obj/target, var/amount=1, var/multiplier=1, var
 	var/part = amount / src.total_volume
 	var/trans_data = null
 	for (var/datum/reagent/current_reagent in src.reagent_list)
+		if (!current_reagent)
```
this is unneeded, a For loop will never return a null value.

Steelpoint commented:
Maybe look to Project Zomboid or NEO Scavenger for a bleeding system.

RemieRichards commented on code/modules/reagents/reagent_containers/blood_pack.dm:
```diff
@@ -0,0 +1,48 @@
+
+	var/blood_type = null
+
+	New()
```
Full typepaths please.

Alek2ander commented:
Blood magic when?

Also, bloodtrails from dragging people should probably be plugged into this.

Maybe also transfer more data to the bloodstains? Detectibes need more love.

RemieRichards commented on code/modules/reagents/reagent_containers/syringes.dm:
```diff
@@ -188,7 +155,16 @@
 				target.add_fingerprint(user)
-				var/trans = src.reagents.trans_to(target, amount_per_transfer_from_this)
+				var/datum/reagent/blood/B
+				for(var/datum/reagent/blood/d in src.reagents.reagent_list)
```
Perhaps change this to ```var/datum/reagent/blood/B = locate() in src.reagents.reagent_list```
it seems silly to write a for loop that will only run one iteration.

TheVekter commented:
Aww fuck, baymed port. Who's ready to die from a single hit from a shard?

:+1: if it's rebalanced to our shit.

RemieRichards commented:
@paprka drip.dmi doesn't seem to be here.

Alek2ander commented:
Also, how will this interact with #6335? Maybe only the SHARP weapons from there should cause bleeding?

dannno commented:
I dearly hope this goes better than goonchem

>ripping up jumpsuits and bedsheets to tie wound

Clothing fab/cloth when

Kromgar commented:
Also its possible to get 40 brute damage as a cultist via the astral projection rune where you become a ghost. You take more damage the longer you are as a ghost

dannno commented:
Or just port the thing from bay where you can write/draw with blood entirely? :))))

paprka commented:
@Kromgar this isn't really my problem, tie your wounds with gauze should you start bleeding

AlexanderUlanH commented:
The obvious solution to cult concerns is to replace brute damage from culty activities with bloodloss.  There're loads of exciting possibilities that could come about from doing that.

paprka commented:
yeah, i'm down for that, cult stealing bloodpacks instead of medibots seems much cooler than just them taking random damage.

GunHog commented:
There should be a MedHUD icon for bleeding and/or low blood status, as that is critical information for a doctor.

paprka commented:
I agree

Commit:
update, gauze, etc

Commit:
Merge branch 'master' of https://github.com/tgstation/-tg-station into ointmentandshit

Conflicts:
	code/game/objects/items/weapons/storage/firstaid.dm

Commit:
adds IV drips

paprka commented:
- Added gauze and improvised gauze
- Gauze is standard issue and will block bleeding for 5 minutes
- Improvised gauze is made by hitting a bedsheet with wirecutters or a shard and blocks bleeding for 2 minutes
- Readded bruise packs and ointments to some medical kits as the primary source of quick healing.
- Nerfed bruise packs to 40 healing instead of 60 brute
- Added a delay for applying bruise packs, ointment, and gauze on yourself (applying it on others is still instant)
- Added a check for THICKMATERIAL so you can't smear ointment on your hardsuit helmet to heal your burns.
- Added the IV drip for blood, which can take/give blood to people at a pretty fast rate. NOTE: please remember to bandage and heal people before giving them transfusions.

No, you can't put acid in blood packs, but you can hook acid beakers up to the IV drip.

I would love for someone to update the wiki on all this stuff later. But le lazy hypocrite paprika hard at work

paprka commented:
BTW I'm removing sleepers in this PR if I get at the map because sleepers are gay and I need that area for blood transfusions and stuff thx

cookingboy3 commented:
I recommend you don't remove sleepers, and if you, at least don't do it in this PR. Hate to see some cool new features get attacked in the PR because it replaced sleepers.

paprka commented:
They already exist, you take toxin damage if you have foreign blood in your system

Commit:
faster bleeding

paprka commented:
- Limbs will now bleed faster (20 brute damage per limb is the threshold for bleeding to begin) because it calculates based on per limb, not total health.

AlexanderUlanH commented:
I don't remember if it's still that way, but last I remembered A was the default blood type, so everyone was type A.  Realistic blood type tolerance probably wouldn't make sense for balance reasons, but you might want to randomize players' blood types if it's going to be important.

Commit:
drip tweak

paprka commented:
No, blood type is randomized every round.

JJRcop commented on code/modules/mob/living/carbon/human/blood.dm:
```diff
@@ -0,0 +1,309 @@
+var/const/BLOOD_VOLUME_SURVIVE = 122
+
+/mob/living/carbon/human/var/datum/reagents/vessel	//Container for blood and BLOOD ONLY. Do not transfer other chems here.
+/mob/living/carbon/human/var/var/pale = 0			//Should affect how mob sprite is drawn, but currently doesn't.
```
```/var/var```, is this supposed to happen?

paprka replied:
procode from paprika

JJRcop commented on code/modules/mob/living/carbon/human/blood.dm:
```diff
@@ -0,0 +1,309 @@
+var/const/BLOOD_VOLUME_BAD = 224
+var/const/BLOOD_VOLUME_SURVIVE = 122
+
+/mob/living/carbon/human/var/datum/reagents/vessel	//Container for blood and BLOOD ONLY. Do not transfer other chems here.
```
Of all the things I could comment about... I think this should be renamed to "circulatory", or something like that.

JJRcop commented:
Can we have a way of draining blood from dead people?

Commit:
fixes dual var

cookingboy3 commented:
New antag type: vampire? Kinda like ling but you have to suck people's blood out.

fj45 commented:
Baycode has vampire...its alright.

dannno commented:
goon has vampire too
even vg might
basically everyone but us

Tenebrosity commented:
Does dragging a body drain blood?

TheVekter commented:
Better question: Does this make it so bodies of people who died from
oxyloss/toxin damage don't cause blood trails?
On Jan 21, 2015 11:51 PM, "homsar" <notifications@github.com> wrote:

> Does dragging a body drain blood?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tgstation/-tg-station/pull/7239#issuecomment-70974793>
> .
>

Tenebrosity commented:
The reason I asked is because it would make the janitor's life a lot easier and fit nicely within the scope of this PR. As would what Vekter suggested.

paprka commented:
Blood smears don't cause bloodloss on bay either and I don't think they should

Tenebrosity commented:
I did some testing.

Issues worth thinking about:
- Dragging bleeding humans who are lying down creates both blood smears and blood drops.
- Smearing your blood on the floor doesn't reduce your blood level.
- Admin rejuvenate verb doesn't reset blood level to 100%.
- Drips of blood erase blood stains they land on (e.g. blood stains from melee attacks, not blood trails)
- You can stick medipens, syringes, brute patches, spray bottles, pills, drink bottles, drink glasses, cups, condiment bottles and condiment packets in the IV drip and they apply the reagents to the mob.

Minor/more nitpicky issues:
- Blood drops on the floor are called "drips of blood" despite their sprites only having only one drop.
- The blood drip description's grammar is wrong: "That's some drips of blood."
- The patient's blood level percentage is shown on the health analyser to 4 dp. No decimals would be cleaner.
- The medical item application messages have repeated the's: "You apply the the medical gauze/ointment/bruise pack/improvised gauze on Salvador Cox." "You begin applying the the improvised gauze on yourself..." "You apply the the improvised gauze on yourself."
- Medical gauze is described as "gauzes" in the stack message: "There are 5 improvised gauzes in the stack."
- There's a blood loss typo in the improvised gauze description: "A roll of cloth roughly cut from something that can stop bloodloss, but does not heal wounds."
- Gauze works irrespective of area targeted.
- The message for your bandage wearing off is unimmersive: "Your bandage wears off." "The blood has soaked through your bandage!" would make more sense
- When someone else is ripped out of an IV, you receive this message: "The needle is ripped out of you, doesn't that hurt?"
-  IV use messages have plain black text instead of user message spanclasses: "Sofia Wood attaches the IV drip to Mark Wahlberg." "There is already a reagent container loaded!"
- When dragging the IV to a patient, the cursor doesn't change to the dragging cursor.
- You can have more than 100% blood. I don't know if this is intentional or whether there's a cut-off.
- There's nothing to indicate somebody is tethered to an IV other than proximity and by examining the IV.
- Monkeys and aliens have no blood

Probably unrelated to this change:
- Admin spawned humans start without a blood type

Idle suggestions:
- Put blood packs in the surgery freezer because it's been around for years and it never gets used
- Change the IV sprite because it skinniness makes it very difficult to click on, it's ugly and it doesn't fit with the art style of the rest of the game

Everything else seems to work fine. I don't know what the balance is like, I'd have to try it on the server.

Steelpoint commented:
A few things I noticed.

* Medkits have Bruise Packs and Ointments again.
* Bleeding has no real affect on healing people at all.

The change is interesting, but during my test, after hitting someone with a crowbar and a energy shield, is that blood loss is of no significant concern.

The patient will regain any minor amount of lost blood once they have been healed. They simply don't lose enough blood for it to be a concern.

There's also no indication of where the person is bleeding from, if anywhere at all, so there's no way to tell where to apply the gauze. However I think the system is such that you just apply the gauze to the patient anywhere on them and it will work.

Maybe going more of the Goon/Bay route of tying health to your blood level would be more appropriate? Since right now blood levels are of little consequence from run of the mill attacks.

-----

It takes a lot of effort to bring someone to very low blood levels (50%), the only way I could do that was continually attacking someone with esword, then healing the brute damage. However once we got that low it becomes very hard to keep the person out of critical as the low blood levels keep them knocked out and suffering form oxyloss, even if I pump chemical into them.

Cyro will keep them alive, but only for so long.

paprka commented:
Give them a transfusion with an IV with a suitable blood pack while healing them.

Steelpoint commented:
I know, I'm just saying what happens if you don't give them a transfusion.

Overall I like it, it just needs to have more of a impact, also bleeding needs to apply to limbs and be detectable.

Either way, I look forward to seeing how this develops.

pudl commented on code/game/objects/items/weapons/storage/firstaid.dm:
```diff
@@ -45,13 +45,13 @@
-	new /obj/item/weapon/reagent_containers/pill/patch/silver_sulf(src)
-	new /obj/item/device/healthanalyzer(src)
+	new /obj/item/stack/medical/gauze(src)
+	new /obj/item/stack/medical/bruise_pack(src)
```
shit

paprka replied:
Uh, no, bruise packs are something doctors rely on, fuck your goonchem being shoved into everything.

pudl commented on code/game/objects/items/weapons/storage/firstaid.dm:
```diff
@@ -102,8 +102,8 @@
 	if (empty) return
-	new /obj/item/weapon/reagent_containers/pill/patch/styptic(src)
-	new /obj/item/weapon/reagent_containers/pill/patch/styptic(src)
+	new /obj/item/stack/medical/gauze(src)
```
remove the brute pack thats in brute medkits and replace it with gauze. i put it there exactly for this

paprka replied:
No

pudl commented:
iv drips are a bad idea, just make the blood pack connect directly to you and transfer blood over 5-10 seconds while you stand still

Delimusca commented:
@pudl what's wrong with IV drips? they're a neat peice of machinery, and they have some interesting alternate uses.

Delimusca commented on code/game/machinery/iv_drip.dm:
```diff
@@ -0,0 +1,164 @@
+	if(src.attached)
+
+		if(!(get_dist(src, src.attached) <= 1 && isturf(src.attached.loc)))
+			attached.visible_message("The needle is ripped out of you, doesn't that hurt?")
```
surely this should be "the needle is ripped out of [attached]", or maybe << ?

paprka replied:
this should only be displaying to the user

pudl commented:
@delimusca they're bad because if you've bled a lot, you're required to sit in one place doing nothing for an extended period of time. that's anti fun. bleeding should be a viable threat that can be easily fixed by a doctor doing their job.

Steelpoint commented:
IV Drips are clumsy. It would be far more effective if it was just a thing that was applied to your arm or backslot for a while.

Delimusca commented:
why not both? also i think on some baycode servers they get auto dragged around with you, either way you could bring it with you, maybe even make them not dense to reduce clutter/shoving.

pudl commented:
a nondense ivdrip that gets autodragged. it would be pointless then. regardless, ivdrips are intifun

paprka commented:
I would prefer if the IV drip was less of a codersprite but I ported it from bay

Odds are you won't be using them much because your heart generates a lot of blood at a decent rate for it to be not annoying

SconesC commented:
somewhat skeptical 'til i see it but i also have a certain degree of faith
a few concerns include the fact that people will die way more often now
mining was also already really really hurt by goonchem medkits and this is probably going to make their bad situation worse

all that being said im sure we can have some fun with the iv drips, also the blood smear sprites are good and should always have been used for bodies

Chiefwaffles commented:
Does surgery cause bleeding until the surgeon clots the bleeders?
If not, it should.

paprka commented:
"mining was also already really really hurt by goonchem medkits and this is probably going to make their bad situation worse"

Well I've reverted goonchem medkits back to bruise packs but you can't apply them through hardsuits.

I honestly really fucking hate mining mobs now to begin with, they end up fucking any change I make and it looks like I'm trying to grudge miners, which I'm not. Mining mobs are just trash gameplay.

Cheridan commented:
@paprka Please don't try to sneak in reverts to other people's works in feature additions. It's poor form.

paprka commented:
Uh, I haven't

Bruise packs and ointments in medkits are part of this feature, they go along with gauze.

Nothing's being snuck in here, I'm pretty sure i stated that I was replacing the shitty patches with what we used to use.

Commit:
some updates

Razharas commented:
I dont like patches for being basically pills(lazy coding) and i dont like the idea of instand health fixing of old ointments and bruisepacks, even with 5 second delay
Instead of combined medbay overhaul this turned into rope pulling game of whos feature will stay in the end

paprka commented:
Goon chem shouldn't be messing with the contents of first aid kits besides replacing chem-based things like pills or syringes, **period.**

The removal of brute packs and ointment was completely unwarranted, lazy, and broke medical doctor by taking away their one reliable way to fix common wounds. Goof has already said it was okay for me to replace the patches where they were, and so I'm going to.

If you want to discuss the removal of brute packs and ointment, do it in a new PR, this revert has already been discussed.

Commit:
updates and optimizations

paprka commented:
Blood drips and blood splatters will now appear on the same tile
Fixed lack of span classes in IV drip
Fixed various grammar
Added a bleeding notification with the health scanner
Healing someone with the admin heal verb will now fill their blood up fully
Dragging someone on the ground will only make blood smears if they are bleeding

Bleeding is not meant to be a significant factor to combat, it's meant to make medical doctors relevant, like defibs do. The idea is if you get into a lot of fights you will lose blood, and even if you patch yourself up with a bruise pack, you will still have a low blood count. You can either wait for your blood to naturally regenerate which takes a long time, eat nutritious food, or do the easiest and fastest thing, get a blood transfusion.

Bleeding and blood punishes people for getting into multiple fights so that they aren't 100% able to heal themselves with just bruise packs afterwards, they will have suffered blood loss which can hurt them in further conflict. Bruise packs are admittedly video-gamey, but they're a necessary and fun part of gameplay that doesn't make the limb targetting completely irrelevant.

I really really really don't want chemicals and viruses and superpowers that will help you produce blood faster, because honestly that will ONCE AGAIN make medical doctor completely fucking irrelevant. In addition to their restored ability to extremely quickly patch common damage types and stop bleeding, they can give blood transfusions and are more likely to be knowledgeable as to which blood type can receive what kind of blood.

Cheridan commented:
If you discussed the change with @Iamgoofball and he's ok with it, then the change is fine. :+1:

Commit:
updates packs

Perakp commented on code/game/machinery/computer/HolodeckControl.dm:
```diff
@@ -230,6 +230,9 @@
 	for(var/obj/effect/decal/cleanable/blood/B in linkedholodeck)
 		qdel(B)
 
+	for(var/obj/effect/decal/cleanable/blooddrip/D in linkedholodeck)
```
Could blooddrip be a subtype of cleanable/blood so you wouldn't need a separate check for them?

paprka replied:
nope

Cheridan replied:
Both checks here and the carp check under it should get torn out, they're frankly ridiculous.

paprka replied:
perhaps

Perakp commented on code/modules/mob/living/carbon/human/human_damage.dm:
```diff
@@ -140,6 +140,10 @@ mob/living/carbon/human/proc/hat_fall_prob()
 	if(update)	update_damage_overlays(0)
 
+/mob/living/carbon/human/proc/restore_blood()
+	if(NOBLOOD in dna.species.specflags)
```
Are you missing a ! here, restoring blood to a species with no blood?

paprka replied:
right yeah

Commit:
fixes noblood check

Perakp commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -29,31 +31,34 @@
 
+		if(stop_bleeding)
+			if(!H.bleedsuppress) //so you can't stack bleed suppression
+				H.suppress_bloodloss(stop_bleeding)
```
You can use gauze to suppress bleeding, before you even start bleeding?

paprka replied:
Yes, just like bruise packs and ointment. Otherwise you could just hit yourself with a gauze without checking if you're actually bleeding.

Perakp replied:
But you can also pre-emptively use gauzes so you won't ever need to worry about bleeding. Each gauze protects for five minutes, so just carry few with you, use one when last one wears off, and you'll never bleed.

paprka replied:
This is really inconsequential powergame that doesn't really matter, and if it does in the future, I'll nerf it to 3 minutes instead

Cheridan replied:
It's not really a powergaming thing, it's a sanity thing. It doesn't make any sense to bandage a wound that doesn't exist to prevent other wounds.

Steelpoint commented:
What precautions are we going to take with antagonists? More appropriately Nuke Ops and Wizards. 

Both antagonists get into a lot of long protracted fights.

Alek2ander commented:
Do any of the ling healy abilities restore blood? Because they should.
The wizard is long overdue for some sort of healing spell. Or, actually, Create food. That could be fun.
Maybe add some sort of chem(s) that block blood loss and/or regenerate blood quickly

Tenebrosity commented:
People stop bleeding after they die, by the way, which doesn't make sense and will presumably stop corpses from smearing blood all together after death. While their current paint brush status is excessive, not bleeding all together is bad too.

On Sat, Jan 24, 2015 at 10:20 PM, paprka <notifications@github.com> wrote:

> Blood drips and blood splatters will now appear on the same tile
> Fixed lack of span classes in IV drip
> Fixed various grammar
> Added a bleeding notification with the health scanner
> Healing someone with the admin heal verb will now fill their blood up fully
> Dragging someone on the ground will only make blood smears if they are bleeding
> Bleeding is not meant to be a significant factor to combat, it's meant to make medical doctors relevant, like defibs do. The idea is if you get into a lot of fights you will lose blood, and even if you patch yourself up with a bruise pack, you will still have a low blood count. You can either wait for your blood to naturally regenerate which takes a long time, eat nutritious food, or do the easiest and fastest thing, get a blood transfusion.
> Bleeding and blood punishes people for getting into multiple fights so that they aren't 100% able to heal themselves with just bruise packs afterwards, they will have suffered blood loss which can hurt them in further conflict. Bruise packs are admittedly video-gamey, but they're a necessary and fun part of gameplay that doesn't make the limb targetting completely irrelevant.
> I really really really don't want chemicals and viruses and superpowers that will help you produce blood faster, because honestly that will ONCE AGAIN make medical doctor completely fucking irrelevant. In addition to their restored ability to extremely quickly patch common damage types and stop bleeding, they can give blood transfusions and are more likely to be knowledgeable as to which blood type can receive what kind of blood.
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/tgstation/-tg-station/pull/7239#issuecomment-71306724

Iamgoofball commented:
@Razharas I made them pill children because pill code had a lot of what I already wanted to do, and I didn't feel like copypasting or reinventing the wheel.

paprka commented:
I'm leaving blood out of dragging blood trails and stuff. Reason being is because you could make someone bleed or put them into crit, then drag them across the station to instakill them on yaketty chase, and then defibbing them would be near impossible

Tenebrosity commented:
Okay. What about making it so when there are blood smears being created, there are no blood droplets? Because drops and smears at the same time looks wrong.

>Dragging someone on the ground will only make blood smears if they are bleeding

Also you should probably make it so blood smears start again after someone dies, because bleeding stops at death.

paprka commented:
i can make blood drops and smears the same color so they don't look so contrasting, that's my bad

paprka commented:
"Also you should probably make it so blood smears start again after someone dies, because bleeding stops at death."

No it doesn't, blood smears still happen regardless

Commit:
updated bloodrips

Commit:
added blood packs and iv drips to cargo

Tenebrosity commented:
Bleeding stopped at death when I first tested this. I assumed if blood smears were tied to bleeding, then they'd stop after the person died.

paprka commented:
Bleeding doesn't stop at death, just drip() processing and stuff. If you were bleeding when you died, you will continue to smear, basically.

RemieRichards commented on code/game/objects/items/devices/scanners.dm:
```diff
@@ -160,6 +160,24 @@ MASS SPECTROMETER
+		if(H.vessel)
+			if(H.blood_max)
+				user.show_message("<span class='danger'>Subject is bleeding!</span>")
+			var/blood_volume = round(H:vessel.get_reagent_amount("blood"))
```
>H:vessel

eww.

paprka replied:
BAYCODE
A
Y
C
O
D
E

JJRcop replied:
For the record I don't blame you for the comma, it's baycode after all.

And yes, a period will work here, since H has already been initiated as a human.

paprka commented:
now that the janicart PR has been merged I"m going to do my map updates here. Gonna put blood packs and the IV drips on the map, so stay tuned and don't merge this.

Commit:
colon and locate() for blood

Commit:
iv drip sprite update

Commit:
Merge branch 'master' of https://github.com/tgstation/-tg-station into ointmentandshit

Commit:
made the iv drip even easier to see on medical tiles

Commit:
adds iv drip and blood packs to the map

paprka commented:
Map changes added, good to go

Steelpoint commented:
Do a changelog before this is merged.

paprka commented:
I'm pretty sure I did oh fuck i didn't commit it

Commit:
changelog and nerf

paprka commented:
changelog added

gauze nerfed to 3 minutes

improvised gauze is 1 1/2 min or 90 seconds

Tenebrosity commented:
But why can you apply gauze to wounds that aren't there?

On Tue, Jan 27, 2015 at 12:58 AM, paprka <notifications@github.com> wrote:

> changelog added
> gauze nerfed to 3 minutes
> improvised gauze is 1 1/2 min or 90 seconds
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/tgstation/-tg-station/pull/7239#issuecomment-71450638

Tenebrosity commented:
It seems like an oversight in the Bay code not to check for bleeding first, as opposed to intended functionality.

On Tue, Jan 27, 2015 at 1:16 AM, Corrie Curtis <raephex@gmail.com> wrote:

> But why can you apply gauze to wounds that aren't there?
> On Tue, Jan 27, 2015 at 12:58 AM, paprka <notifications@github.com> wrote:
>> changelog added
>> gauze nerfed to 3 minutes
>> improvised gauze is 1 1/2 min or 90 seconds
>> ---
>> Reply to this email directly or view it on GitHub:
>> https://github.com/tgstation/-tg-station/pull/7239#issuecomment-71450638

Stickymayhem commented:
I'm really scared by this. I have hopes for it but this and goonchem have been pushing the game from an antagonist's perspective towards attrition being the main cause of death.

The protracted murderbones that used to be possible seem to be becoming less and less likely, and everyone on the station now seems limited to bursts of destruction or heroism. You can still disarm that nuke op, but killing every single one solo is going to wear you out unless you have the time to be juggling your various concerns and wield extremely thorough knowledge of interacting systems. 

I think this is going to make the game frustrating for people who don't understand these mechanics, but that's already a problem in general so I HOPE overall this will be an improvement. It always seemed silly both from a realistic and game balance perspective that someone with the right set up could essentially destroy the station stacking up colossal amounts of damage with no ill effects.

I'm also a big fan of making one of the most boring jobs on the station be more needed so I'm interested by this change but I think it's worth noting the direction the game is taking as a result of it.

paprka commented:
I got some feedback from this and did some test rounds with friends. I dunno if we should end up implementing this, I'd rather have a player ingame poll first maybe.

Lumipharon commented:
I'd ideally prefer a slightly simpler system, for example it seems unintuitive that blunt force trauma is going to cause loads of bleeding, just because it did a lot of brute damage.

Just make sharp things have a chance to cause bleeds (of varying severity), that just cause damage over time - more straightforwards then worrying about damage AND bloodloss.

Then bandages (that stack like bruisepacks), and ripping fabric for makeshift bandages, and bloodclottazine (tm) for chemist to make bleed stopping patches. And bleeding could heal over time (if bandaged), or there could be ghetto shit like sowinng wounds up with wire/using a welding tool to cauterise the wound etc etc.

Basically the big thing I think, is to make it managable without dedicated medical equipment, but not ideal.

paprka commented:
the way bleeding is caused is super, super easily changed. Like really easy. you can just call drip() if you want to take blood from people with sharp weapons. remie needs to implement dismemberment before that can happen.

ghetto sewing your wounds up with wire sounds really cool. I'd be down for that as long as it gave you some more brute damage or something from the pain, i dunno.

pudl commented:
bleeding, imo, shouldn't be a major gameplay aspect. it should simply be another status effect.
for example, if something causes you to have losebreath, you don't need to go to medbay and do some hardcoded thing for 5 minutes as if it's some 100% special damage.
bleeding should just be caused by sharp objects (after rr finishes her pr), and as you lose blood you have status effects build until you drop dead at 0 units. it should be fixed easily, as well. for example, gauze can make bleeding slowly peter down until you've stopped, and at medbay you can find sutures which immediately stop bleeding, possibly with 5-10 brute from the pain of the suture.
i really hate iv drips, again, the problem is that baymed is very very in depth because it's rp oriented. gauzing/suturing your wound, attaching a full pack to yourself, standing still for 10-15 seconds, then being fine to go would be a good system that still is nothing to joke about,.

paprka commented:
i don't like the idea of rpg-style status effects. I just want this to be semi-realistic as most things usually are while still remaining fun. I don't want it to be another way to powergame by using sharp weapons instead of blunt weapons, because blunt weapons can cause bleeding too. I would prefer if bleeding just punished people who self-healed religiously instead of visiting medbay. Medical doctors need relevance.

pudl commented:
ah i don't mean rpg-style status effects. i mean as you lose blood you walk slower, your eyesight gets blurry, just minor annoyances to tell you "get to medbay!"
i understand the notion of sharp weapons being better for powergaming, but on that note it may be good for sharp weapons that cause bleeding to be low-blunt damage causing. so a scalpel may cause bleeding but only 5 blunt, while a toolbox won't cause someone to bleed but it does a heck of a lot more damage.
if anything can you at least add sutures to medbay? i think they would be a good reason to go to medbay for help as opposed to the dorms where you can rip up bedsheets.

paprka commented:
What do sutures even do on baycode, and what makes them different from gauze exactly

pudl commented:
i don't know if baycode has sutures. i'm just saying i think that a good idea would be for gauze to be a slower but easier way to handle bleeding, while sutures give you a bit of brute but instantly stops all bleeding (which may be more appealing to the risk of bleeding out while gauze slowly stops bleeding)

JJRcop commented:
Blunt weapons can easily break skin if used hard enough. Also gameplay > realism every time.

pudl commented:
being able to bleed out from someone toolboxing you is shitty

JJRcop commented:
subjective, I don't think it is

AnonymousNow commented:
> I don't have a git account but Pap if you see this thread maybe consider cauterisation as a way to ghetto stop bleending. Slight burn damage in exchange for not loosing your precious fluids  - Wyzack

I'm glad to see bruise packs & ointment return, but I worry somewhat for the autonomy of spessmen.

paprka commented:
Yeah JJRcop but this stuff can be changed so that sharp weapons trigger it upon impact, but again, this will have to happen later on after remie's stuff is merged

paprka commented:
Gee guess what PR is going to go through 50000000000000000000000000000000000000000000000000000000000000000000000000000000000 conflict fixes before it's EVER reviewed

this one

Dorsisdwarf commented:
This is the most commented on PR I've ever seen.

My eyes are bleending

Nyvrem commented:
Vomiting blood should cause the person that vomits blood loss.

Commit:
Merge branch 'master' of https://github.com/tgstation/-tg-station into ointmentandshit

Conflicts:
	_maps/map_files/TgStation/tgstation.2.1.3.dmm

Commit:
map conflicts and other changes

Commit:
removes carp and blood checks from holodeck

Commit:
fixes some gauze inconsistencies

paprka commented:
You can't apply gauze anymore if someone isn't bleeding or they already have gauze on their bleeding wound. @Cheridan I also removed those shitty checks from the holodeck.

Commit:
fixes blood drips

Commit:
finally actually fixes the blood decals

paprka commented:
Ok fucking review time please. :cake:

paprka commented:
R E V I E W
E
V
I
E
W

also here's a pic of updated medical storage:

![ss 2015-02-01 at 05 36 52](https://cloud.githubusercontent.com/assets/7247232/5994875/ec7b12da-aa38-11e4-9f7d-a07aa38dc666.png)

Dorsisdwarf commented on code/game/objects/structures/bedsheet_bin.dm:
```diff
@@ -32,6 +32,12 @@ LINEN BINS
 	return
 
+/obj/item/weapon/bedsheet/attackby(obj/item/I, mob/user)
+	if(istype(I, /obj/item/weapon/wirecutters) || istype(I, /obj/item/weapon/shard))
```
What about knives? Shhouldn't they work?

And what about energy swords - you can cut cake with them, after all.

paprka replied:
honestly this is just placeholder until remie's PR is done so we can cut all this snowflake checking out and actually use the updated sharp stuff.

optimumtact commented:
Have you thought about burns needing gauze as well?

paprka commented:
No because stupid ideas don't enter my brain

optimumtact commented:
Think about it then

paprka commented:
No. Also someone review this before it rots and things break.

Cheridan commented:
:bear: 
:bikini: 
:high_heel:

Anonus commented:
it's happening

PKPenguin321 commented on code/game/machinery/iv_drip.dm:
```diff
@@ -0,0 +1,164 @@
+	if(src.attached)
+
+		if(!(get_dist(src, src.attached) <= 1 && isturf(src.attached.loc)))
+			attached << ("<span class='warning'>The IV drip needle is ripped out of you, doesn't that hurt?</span")
```
instead of making an issue report i am going to try to into github to tell you that you didn't close this </span> properly so the message displays incorrectly in-game
hopefully you see this and it works

Tenebrosity replied:
Also that warning message is pretty shit

On Wed, Feb 4, 2015 at 3:20 PM, PKPenguin321 <notifications@github.com>
wrote:

>> +		W.loc = src
>> +		src.beaker = W
>> +		user << "You attach \the [W] to \the [src]."
>> +		src.update_icon()
>> +		return
>> +	else
>> +		return ..()
>> +
>> +
>> +/obj/machinery/iv_drip/process()
>> +	set background = 1
>> +
>> +	if(src.attached)
>> +
>> +		if(!(get_dist(src, src.attached) <= 1 && isturf(src.attached.loc)))
>> +			attached << ("<span class='warning'>The IV drip needle is ripped out of you, doesn't that hurt?</span")
> instead of making an issue report i am going to try to into github to tell you that you didn't close this </span> properly so the message displays incorrectly in-game
> hopefully you see this and it works
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/tgstation/-tg-station/pull/7239/files#r24058596

Miauw62 commented:
Ayy lmao prepare for the complaining as we get yet another misplaced
dysfunctional medical overhaul.
Op 4-feb.-2015 02:23 schreef "Anonus" <notifications@github.com>:

> it's happening
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tgstation/-tg-station/pull/7239#issuecomment-72772682>
> .
>

Steelpoint commented:
Welcome to /tg/code, where we roll with it.
